prom2json.go

```diff
@@ -145,30 +143,30 @@ func FetchMetricFamilies(
 		}
 	}
 	client := &http.Client{Transport: transport}
-	decodeContent(client, url, ch)
+	return decodeContent(client, url, ch)
 }
 
-func decodeContent(client *http.Client, url string, ch chan<- *dto.MetricFamily) {
+func decodeContent(client *http.Client, url string, ch chan<- *dto.MetricFamily) error {
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		log.Fatalf("creating GET request for URL %q failed: %s", url, err)
+		return fmt.Errorf("creating GET request for URL %q failed: %s", url, err)
 	}
 	req.Header.Add("Accept", acceptHeader)
 	resp, err := client.Do(req)
 	if err != nil {
-		log.Fatalf("executing GET request for URL %q failed: %s", url, err)
+		return fmt.Errorf("executing GET request for URL %q failed: %s", url, err)
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		log.Fatalf("GET request for URL %q returned HTTP status %s", url, resp.Status)
+		return fmt.Errorf("GET request for URL %q returned HTTP status %s", url, resp.Status)
```

Comment:
See above.

Reply:
`resp.Status` returns a string.
I can replace `%s` by `%v` but not sure ir make sense here.

WDYT @beorn7?

Reply:
Yes, my bad.

Reply:
(But just FYI: `%v` used on a string renders the string the same way as `%s`. I often just use %v because it is more flexible.)